src/qt/optionsmodel.cpp

```diff
@@ -53,12 +54,16 @@ void OptionsModel::Init(bool resetSettings)
     // These are Qt-only settings:
 
     // Window
-    if (!settings.contains("fHideTrayIcon"))
+    if (!QSystemTrayIcon::isSystemTrayAvailable()) {
+        settings.setValue("fHideTrayIcon", true);
```

Comment:
Kinda of nasty to change the saved setting just because you run it once on a platform that doesn't support it. I suggest leaving it be in settings, greying the setting, and ignoring it.